### PR TITLE
adding Dockerfile 

### PR DESCRIPTION
**Pull request by romaissaMe**
@achab

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -7,6 +8,5 @@ openai==0.27.8
 python-multipart==0.0.6
 fastapi==0.97.0
 uvicorn==0.22.0
-python-dotenv==1.0.0
+python-dotenv
```
Comment (achab) [inline requirements.txt:11]: please add the version here

File: requirements.txt [hunk 2]
```diff
@@ -7,6 +8,5 @@ openai==0.27.8
...
 requests~=2.31.0
-https://github.com/explosion/spacy-models/releases/download/en_core_web_lg-3.5.0/en_core_web_lg-3.5.0.tar.gz
```
Comment (achab) [inline requirements.txt:12]: why removing it?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,13 @@
+FROM python:3.9.7
+
+WORKDIR /usr/src/application
+
+COPY requirements.txt ./
```
Comment (achab) [inline Dockerfile:5]: you can put `COPY requirements.txt requirements.txt` instead

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+RUN pip install --no-cache-dir -r requirements.txt
+
+RUN python -m spacy download en_core_web_lg
```
Comment (achab) [inline Dockerfile:9]: this one is not necessary if the model is added to requirements.txt

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+COPY . .
+
+CMD [ "uvicorn","app.api.main:app","--host","0.0.0.0"]
```
Comment (achab) [inline Dockerfile:13]: please add whitespaces here:

```bash
CMD ["uvicorn", "app.api.main:app", "--host", "0.0.0.0"]
```

File: requirements.txt [hunk 6]
```diff
@@ -1,3 +1,4 @@
+pip==23.1.2
```
Comment (achab) [inline requirements.txt:1]: why? I didn't know that it works

File: requirements.txt (hunk 1 above)
Comment (romaissaMe) [reply requirements.txt:11]: in some hosting platform i got the error of not finding python-dotenv==1.0.0 however is the correct version I have, so I got rid of it :)

Comment (achab) [reply requirements.txt:11]: ok, it's pretty weird

File: requirements.txt (hunk 2 above)
Comment (achab) [reply requirements.txt:12]: what is "lg"?

File: .gitignore [hunk 7]
```diff
@@ -5,3 +5,6 @@ __pycache__
 .vscode
 uploads
 
+
+.space
+.vercel
```
Comment (achab) [inline .gitignore:10]: just add the platform we will keep, so probably render. you can remove folders of platforms we won't use

File: Dockerfile
Comment (achab) [inline Dockerfile:1]: a possible next task will be to create a github action to build the docker image and publish it

**Pull request closed without merging**